src/concrete/ml/pytest/utils.py

```diff
@@ -127,7 +129,13 @@ def _get_sklearn_models_and_datasets(model_classes: List, unique_models: bool =
             # the test execution timings
             n_classes_to_test = (
                 [2]
-                if unique_models or get_model_class(model_class) == KNeighborsClassifier
+                if unique_models
+                or get_model_class(model_class) == KNeighborsClassifier
+                or (
+                    isinstance(model_class, partial)
+                    and model_class.func == SGDClassifier
+                    and model_class.keywords.get("fit_encrypted", False)
```

Comment:
I don't get why it's working : shouldn't it be the opposite ? iirc, FHE training does not handle other than binary classification, but here we are puting 2 classes if *not* using FHE training (`and model_class.keywords.get("fit_encrypted", False)`), while 2 and 4 with FHE training 🤔 It should fail with 4 classes no ?

Reply:
> if not using FHE training (and model_class.keywords.get("fit_encrypted", False))

`.get(key, default)` returns the value for the key. If key doesn't exist it returns the default so False. So should be good, right? But yeah it's confusing.

We could write

```python
    and "fit_encrypted" in model_class.keywords
    and model_class.keywords["fit_encrypted"]
```

instead.


Note that this is useless because of this:

https://github.com/zama-ai/concrete-ml/blob/4f759790ab77d25f296bfa41626b3fa0a80021bc/src/concrete/ml/pytest/utils.py#L200-L201

Reply:
ah yes my bad of course, read too fast. But yes you could write `model_class.keywords["fit_encrypted"]` instead since we are checking that we have a `SGDClassifier` model first anyway !